### PR TITLE
OES-33 delete users functionality in openam

### PR DESCRIPTION
**Pull request by sneha-nagole**
# Summary
Expose new method to delete users by uid from OpenAm

# What's New
New method to delete users by uid from OpenAm

# What is changed

### CODE REVIEW

File: openam-tools/src/main/java/edu/mit/ll/soa/sso/util/OpenAmBaseUtil.java [hunk 1]
```diff
@@ -567,6 +567,50 @@ protected AMIdentity createUserIdentity(String email, String pass,
...
+        IdSearchResults results = idRepo.searchIdentities(IdType.USER, "*", searchControl);
+        Set ids = results.getSearchResults();
+        return ids;
+    }
+
+    public boolean deleteUserIdentity(String uid) throws Exception {
+        System.out.println("\ndeleteIdentity: TypeID: " + IdType.USER + ", ID: " + uid + "\n");
+        try {
+            if (idRepo == null) {
+                System.out.println("idRepo is null!");
```
Comment (comeara) [inline openam-tools/src/main/java/edu/mit/ll/soa/sso/util/OpenAmBaseUtil.java]: Please change System.out.println calls to debug log entries or remove.

File: openam-tools/src/main/java/edu/mit/ll/soa/sso/util/OpenAmBaseUtil.java [hunk 2]
```diff
@@ -567,6 +567,50 @@ protected AMIdentity createUserIdentity(String email, String pass,
 		return null;
 	}
 
+    private Set findIdentityByUID(String uid) throws Exception {
```
Comment (comeara) [inline openam-tools/src/main/java/edu/mit/ll/soa/sso/util/OpenAmBaseUtil.java]: You can use your discretion as to whether you want to incorporate this or not but here is a more succinct and type-safe implementation.

```
	private Set findIdentityByUID(String uid) throws Exception {
		Map<String, Set<String>> criteria = Collections.singletonMap("uid", Collections.singleton(uid));
		IdSearchControl searchControl = new IdSearchControl();
		searchControl.setSearchModifiers(IdSearchOpModifier.AND, criteria);
		return idRepo.searchIdentities(IdType.USER, "*", searchControl).getSearchResults();
	}
```

**Review by DreamWver — APPROVED** [review]

File: sso-tools/src/main/java/edu/mit/ll/nics/sso/util/SSOUtil.java [hunk 3]
```diff
@@ -263,6 +263,18 @@ public String createUser(String email, String pass, String first, String last, b
...
+    public String deleteUser(String uid) {
+        boolean status = false;
+        try {
+          status = utils.deleteUserIdentity(uid);
+        } catch(Exception e){
+            log.error("Error deleting user identity by uid : " + uid, e);
+        }
+        String message = status ? "successfully deleted identity with uid : " + uid : "failed to delete identity with uid : " + uid;
+
+        return "{\"status\":\"" + ((status) ? "success" : "fail") + "\",\"message\":\"" + message + "\"}";
```
Comment (comeara) [inline sso-tools/src/main/java/edu/mit/ll/nics/sso/util/SSOUtil.java:275]: I see how you were following an existing pattern here and I appreciate that. Returning a string that can be parsed into JSON is not the kind of design we want to propagate in the project.

**Review by comeara — CHANGES_REQUESTED** [review]
Good job working in a tough code base.

Functionally, I think this is fine and follows the patterns established in the code.

Mid-to-long term, I want to get away from Util classes like this. We want to get to the place where we are not adding methods to classes with 1000+ lines of code.

**Review by comeara — APPROVED** [review]
👍

**Pull request merged**